### PR TITLE
mailroom3.py

### PR DESCRIPTION
**Pull request by JerHsi**

### CODE REVIEW

File: students/JerryH/Lesson05/mailroom3.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+def create_report():
+    print("\nDonor Name           |  Total Given | Num Gifts | Average Gift")
+    print("---------------------------------------------------------------\n")
+    report = []  # initialize report
+
+    # for each_name in get_all_donor_names(donors):
+    for each_donor in donors.keys():
+        report.append([each_donor, sum(donors[each_donor]), len(donors[each_donor])])
```
Comment (natasha-aleksandrova) [inline students/JerryH/Lesson05/mailroom3.py]: you could replace this section with list comprehension:
```
report = [[dk, sum(dv), len(dv)] for dk, dv in donors.items()]

```

File: students/JerryH/Lesson05/mailroom3.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        donor_name = donor_name_prompt()
+        if donor_name.lower() == "list":
+            donor_name = None
+            list_all_donor_names()
+
+    donation = None
+    while not donation:
+        try:
+            donation = donation_prompt()
+        except:
```
Comment (natasha-aleksandrova) [inline students/JerryH/Lesson05/mailroom3.py]: because of the exception `donation` will not have a value assigned so no need to set `donation = None`

**Comment by natasha-aleksandrova** [comment]
@JerHsi I don't see your submission on the course website, could you please submit it?

**Comment by JerHsi** [comment]
@natasha-aleksandrova just sent it.

**Pull request merged**